### DISCUSSION TITLE
[Support] 在目前版本中，通过@EventSubscriberModel订阅sql执行记录时，在被这个注解标注的类中可以获取到请求的上下文对象吗

### DISCUSSION THREAD

**Discussion by justcode-cc** [midwayjs/midway#4081] [Q&A] [answered]
### Describe the problem（描述问题）

如下所示，在使用了@EventSubscriberModel()标记的类型，无法直接通过ctx: Context;获取到http请求的上下文。

`
@EventSubscriberModel()
export class SqlSubscriber implements EntitySubscriberInterface<any> {

  @Inject('ctx')
  ctx: Context; //这里的ctx是空的

  beforeInsert(event: InsertEvent<any>) {
    // 拦截插入操作的 SQL
    console.log('Before Insert:', event.queryRunner.query);
  }

  beforeUpdate(event: UpdateEvent<any>) {
    // 拦截更新操作的 SQL
    console.log('Before Update:', event.queryRunner.query);
  }

  beforeRemove(event: RemoveEvent<any>) {
    // 拦截删除操作的 SQL
    console.log('Before Remove:', event.queryRunner.query);
  }

  beforeQuery(event: BeforeQueryEvent<any>): Promise<any> | void {
    // this.ctx.logger.info('sql:' + event.query+' ; param:' + event.parameters)
  }

  afterQuery(event: AfterQueryEvent<any>): Promise<any> | void {
  }
}
`

### Midway Versions（Midway 版本）

midway-version@1.4.0

**Comment by czy88840616** [comment]
这个类是由 typeorm 控制，不在框架的控制范围内，所以没有 ctx

**Reply by justcode-cc** [reply]
有其他折中的办法吗，在打印sql日志时，获取到当前的上下文

**Reply by czy88840616** [reply, marked as answer]
有

1、启用 https://github.com/midwayjs/midway/blob/main/packages/tenant/src/configuration.ts#L9
2、获取 ctx https://github.com/midwayjs/midway/blob/main/packages/tenant/src/tenantManager.ts#L24

**Reply by justcode-cc** [reply]
非常感谢，已搞定
